### DISCUSSION TITLE
[RESOLVED] Difference between Method.hasName("XXX") and Method.getName() == "XXX"?

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [github/codeql#7694] [General]
Is there any difference between `Method.hasName("XXX")` and `Method.getName() == "XXX"`?

**Comment by intrigus-lgtm** [comment]
Looking at the source code shows that there is no difference between the two:
https://github.com/github/codeql/blob/70f4efb8343f9a96912d3f5c9e3aa6343ba69a17/java/ql/lib/semmle/code/java/Element.qll#L11-L15

There might potentially be a minor performance difference, but I'm not sure about that. Which ever form you use is fine.
